src/components/ecl-button/ecl-button.tsx

```diff
@@ -19,40 +19,74 @@ export class EclButton {
   @Prop() hideLabel: boolean = false;    
   @Prop() ariaControls: string;
   @Prop() itemId: string;
+  @Prop() showIndicator: boolean = false;
+  @Prop() indicatorValue: string = '';
+  @State() indicatorSlot: 'before' | 'after' = 'after';
```

Comment:
something i don't understand, having said that it's not understandable the reason why the indicator depends on an icon, if that is the case then why do we need to "calculate" the indicatorSlot, isn't that just simply tied to hasIconBefore or hasIconAfter..? Just for clarity and simplification...

Reply:
refactored the button with your feedback